03_inheritance/03_02_inheritance.py

```diff
@@ -13,22 +14,33 @@
 
 class Movie:
     """Movies"""
-    def __init__ (self, title, year):
+    def __init__ (self, title, year, genre, length, rating):
         self.title = title
         self.year = year
+        self.genre = genre
+        self.length = length #in minutes
+        self.rating = rating
 
     def __str__(self) -> str:
         return f"{self.title} {self.year}"
 
     
+class RomCon(Movie):
+    """Romantic Comedies"""
+    def __init__(self, title, year, length, rating, heart_throb, genre= "romantic comedy"):
+        super().__init__(title, year, genre, length, rating)
+        self.heart_throb = heart_throb
+
 class ActionMovie (Movie):
     """Action Movies"""
 
-    def __init__(self, title, year, pg=13):
-        super().__init__(title, year)
+    def __init__(self, title, year, genre, length, rating, franchise, pg=13):
+        super().__init__(title, year, genre, length, rating)
         self.pg = pg
+        self.franchise = franchise
 
+m = ActionMovie("Karate Kid", 1981, "action", 91, "PG-13", "Karate Kid")
+n = RomCon("You've Got Mail", 1998, 102, 13, "Tom Hanks")
```

Comment:
**suggestion**: When dealing with a bunch of arguments like this, it's usually better to use keyword arguments:

```suggestion
m = ActionMovie(
    title="Karate Kid",
    year=1981,
    genre="action",
    length=91,
    rating="PG-13",
    franchise="Karate Kid",
)
n = RomCon(
    title="You've Got Mail", year=1998, length=102, rating=13, heart_throb="Tom Hanks"
)
```

This makes things more redable and also explicitly shows how instantiating the two classes require different arguments. With positional arguments, as you've used, it quickly becomes confusing what refers to what, and you have to look back at the class definition to remember what the arguments do.

The other nice thing about keyword arguments for these use cases is that you can mix up the order of the arguments:

```python
m = ActionMovie(
    length=91,
    rating="PG-13",
    franchise="Karate Kid",
    title="Karate Kid",
    year=1981,
    genre="action",
)
```

Note how length is first and title is in the middle, even though in the `__init__()` function they are in a totally different order, and it will still work.